app/src/main/java/com/example/movies/ui/authorization/AuthorizationFragment.kt

```diff
@@ -0,0 +1,135 @@
+package com.example.movies.ui.authorization
+
+import android.annotation.SuppressLint
+import android.graphics.Bitmap
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebChromeClient
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.NavController
+import androidx.navigation.Navigation
+import androidx.navigation.fragment.findNavController
+import com.example.data.cache.RequestTokenDataCache
+import com.example.movies.R
+import com.example.movies.databinding.FragmentAuthorizationBinding
+import com.example.movies.utils.AUTH_URI
+import com.example.movies.utils.LOGOUT_URL_IN_WEB_VIEW
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class AuthorizationFragment : Fragment() {
+
+    private var _binding: FragmentAuthorizationBinding? = null
+    private val binding: FragmentAuthorizationBinding
+        get() =
+            _binding ?: throw RuntimeException("FragmentAuthorizationBinding == null")
+
+    private val viewModel: AuthorizationViewModel by viewModels()
+
+    @Inject
+    lateinit var requestTokenDataCache: RequestTokenDataCache
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentAuthorizationBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        openWebPageForAuthorization(savedInstanceState)
+    }
+
+    private val mWebViewClient: WebViewClient by lazy {
+        object : WebViewClient() {
+            override fun shouldOverrideUrlLoading(
+                view: WebView?,
+                request: WebResourceRequest?
+            ): Boolean {
+                val url = request?.url.toString()
+
+                if (url == LOGOUT_URL_IN_WEB_VIEW) {
+                    viewModel.clearCookies()
+                    viewModel.deleteTokenFromCache()
+                    viewModel.getRequestToken()
+                    return true
+                }
+
+                view?.loadUrl(url)
+
+                return true
+            }
+
+            override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
+                if (viewModel.isAuthorizationApproved(url?:"")) {
+                    viewLifecycleOwner.lifecycleScope.launch {
```

Comment:
Do you need to launch coroutine here?
Fragment could just call viewmodel's method in usual non-coroutine code, and ViewModel will launch coroutine to run some usecase.

Reply:
Fixed